definition/command/ICommandUtility.ts

```diff
@@ -0,0 +1,50 @@
+import {
+    IHttp,
+    IModify,
+    IPersistence,
+    IRead,
+} from "@rocket.chat/apps-engine/definition/accessors";
+import { IRoom } from "@rocket.chat/apps-engine/definition/rooms";
+import { IUser } from "@rocket.chat/apps-engine/definition/users";
+import { AiGifApp } from "../../AiGifApp";
+import { RequestDebouncer } from "../../src/helper/RequestDebouncer";
+import {
+    ISlashCommandPreview,
+    ISlashCommandPreviewItem,
+} from "@rocket.chat/apps-engine/definition/slashcommands";
+
+export interface ICommandUtility {
+    app: AiGifApp;
+    params: Array<string>;
+    sender: IUser;
+    room: IRoom;
+    read: IRead;
+    modify: IModify;
+    http: IHttp;
+    persis: IPersistence;
+    triggerId?: string;
+    threadId?: string;
+    requestDebouncer: RequestDebouncer;
+
+    resolveCommand(): Promise<ISlashCommandPreview>;
+    resolveExecutePreviewItem(item: ISlashCommandPreviewItem): Promise<void>;
+}
+
+export interface IPreviewerUtilityParams {
+    app: AiGifApp;
+    params: Array<string>;
+    sender: IUser;
+    room: IRoom;
+    read: IRead;
+    modify: IModify;
+    http: IHttp;
+    persis: IPersistence;
+    triggerId?: string;
+    threadId?: string;
+    requestDebouncer: RequestDebouncer;
+}
```

Comment:
@AmanNegi we can apply here as well in some way, try that out.

Reply:
Okay I'll simplify it, i don't believe we have the need for all those params in the `ICommandUtility` class.